WireAPI/Sources/WireAPI/APIs/UserClientsAPI/UserClientsAPIV0.swift

```diff
@@ -0,0 +1,186 @@
+
+        guard let url = components?.url else {
+            assertionFailure("generated an invalid url")
+            throw UserClientsAPIError.invalidURL
+        }
+
+        let body = try JSONEncoder.defaultEncoder.encode(
+            UserClientsRequestV0(qualifiedIDs: Array(userIDs))
+        )
+
+        let request = URLRequestBuilder(url: url)
+            .withMethod(.post)
+            .withBody(body, contentType: .json)
+            .build()
+
+        let (data, response) = try await apiService.executeRequest(
+            request,
+            requiringAccessToken: true
+        )
+
+        return try ResponseParser()
+            .success(code: .ok, type: OtherUserClientsV0.self)
+            .parse(code: response.statusCode, data: data)
+    }
+}
+
+struct ListUserClientV0: Decodable, ToAPIModelConvertible {
+
+    let payload: [SelfUserClientV0]
+
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let payload = try container.decode([SelfUserClientV0].self)
+        self.payload = payload
+    }
+
+    func toAPIModel() -> [SelfUserClient] {
+        payload.map { $0.toAPIModel() }
+    }
+}
+
+struct SelfUserClientV0: Decodable, ToAPIModelConvertible {
+
+    let id: String
+    let type: UserClientType
+    let activationDate: UTCTimeMillis
+    let label: String?
+    let model: String?
+    let deviceClass: DeviceClass?
+    let lastActiveDate: UTCTime?
+    let mlsPublicKeys: MLSPublicKeys?
+    let cookie: String?
+    let capabilities: CapabilitiesList?
+
+    struct CapabilitiesList: Decodable {
+
+        let capabilities: [UserClientCapability]
+
+    }
+
+    enum CodingKeys: String, CodingKey {
+
+        case id
+        case type
+        case activationDate = "time"
+        case label
+        case model
+        case deviceClass = "class"
+        case lastActiveDate = "last_active"
+        case mlsPublicKeys = "mls_public_keys"
+        case cookie
+        case capabilities
+
+    }
+
+    func toAPIModel() -> SelfUserClient {
+        SelfUserClient(
+            id: id,
+            type: type,
+            activationDate: activationDate.date,
+            label: label,
+            model: model,
+            deviceClass: deviceClass,
+            lastActiveDate: lastActiveDate?.date,
+            mlsPublicKeys: mlsPublicKeys,
+            cookie: cookie,
+            capabilities: capabilities?.capabilities ?? []
+        )
+    }
+
+}
+
+struct UserClientsRequestV0: Encodable {
+
+    let qualifiedIDs: [UserID]
+
+    enum CodingKeys: String, CodingKey {
+        case qualifiedIDs = "qualified_users"
+    }
+
+}
+
+struct OtherUserClientsV0: Decodable, ToAPIModelConvertible {
+    typealias Domain = String
+    typealias UserID = String
+
+    let qualifiedUserMap: [Domain: [UserID: [OtherUserClient]]]
```

Comment:
We should be careful not to rely on the decoding shape of public api models. The risk is that the public model changes which inadvertently breaks the decoding of this network response.

In fact, we really shouldn't expose public api models such as `OtherUserClient` as `Codable` since we allow unknown usage of `Codeable` (e.g encode a model to data, store in db) which could easily break with a internal change.

For this reason I think we should keep encoding / decoding strictly local.

Reply:
It makes sense 👍 fixed in https://github.com/wireapp/wire-ios/pull/2019/commits/9b95e7ed9c23bc7b9d0509f78a375282948b3a29 

note: outside of the scope of this PR but I think we also need to revisit some of our public models that are currently Codable to align with that approach

Reply:
Yes we already have a ticket for it